hoomd/mesh.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Implement Mesh."""
+
+import hoomd
+from hoomd import _hoomd
+from hoomd.operation import _HOOMDBaseObject
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.data.typeconverter import OnlyIf, to_type_converter
+from collections.abc import Sequence
+from hoomd.logging import log
+import numpy as np
+
+
+class Mesh(_HOOMDBaseObject):
+    """Data structure combining multiple particles into a mesh.
+
+    The mesh is defined by an array of triangles tht make up a
+    triangulated surface.
+
+    Args:
+        name ([`str`]): name of the mesh that also acts as a
+        type name. Only one type per mesh can be defined!
+
+    Examples::
+
+        mesh = mesh.Mesh()
+        mesh.triangles = [[0,1,2],[0,2,3],[0,1,3],[1,2,3]]
+
+    """
+
+    def __init__(self, name=["mesh"]):
+
+        param_dict = ParameterDict(size=int,
+                                   types=OnlyIf(
+                                       to_type_converter([
+                                           str,
+                                       ] * 1),
```

Comment:
```suggestion
                                   types=OnlyTypes([str], postprocessing=lambda x: [x], strict=True),
```
You can't use this directly, but this is more what you want, I think.

Reply:
You may have to keep preprocessing; I am not sure if querying from C++ with this will error or not.

You could also just accept the single string and configure the C++ side that way.

Or if this is to enable more types later, than we should leave as is and close this suggestion. Use a tuple in the `name` argument then.